DB/src/main/java/io/deephaven/db/v2/utils/ProcessMemoryTracker.java

```diff
@@ -0,0 +1,99 @@
+package io.deephaven.db.v2.utils;
+
+import io.deephaven.configuration.Configuration;
+import io.deephaven.db.tablelogger.ProcessMemoryLogLogger;
+import io.deephaven.db.v2.QueryTable;
+import io.deephaven.internal.log.LoggerFactory;
+import io.deephaven.io.logger.Logger;
+
+import java.io.IOException;
+
+public class ProcessMemoryTracker {
+    private static final long REPORT_INTERVAL_MILLIS = Configuration.getInstance().getLongForClassWithDefault(
+            ProcessMemoryTracker.class, "reportIntervalMillis", 15 * 1000L);
+
+    private static volatile ProcessMemoryTracker INSTANCE;
+    private static boolean started = false;
+
+    public static ProcessMemoryTracker getInstance() {
+        if (INSTANCE == null) {
+            synchronized (ProcessMemoryTracker.class) {
+                if (INSTANCE == null) {
+                    INSTANCE = new ProcessMemoryTracker();
+                }
+            }
+        }
+        return INSTANCE;
+    }
+
+    private final Logger logger;
+
+    private final MemoryTableLogger<ProcessMemoryLogLogger> processMemLogger;
+
+    private ProcessMemoryTracker() {
+        logger = LoggerFactory.getLogger(UpdatePerformanceTracker.class);
+        processMemLogger = new MemoryTableLogger<>(
+                logger, new ProcessMemoryLogLogger(), ProcessMemoryLogLogger.getTableDefinition());
+    }
+
+    private void startThread() {
+        Thread driverThread = new Thread(new ProcessMemoryTracker.Driver(), "ProcessMemoryTracker.Driver");
+        driverThread.setDaemon(true);
+        driverThread.start();
+    }
+
+    public static void start() {
+        synchronized (UpdatePerformanceTracker.class) {
```

Comment:
Wrong lock.

Also, I think you should just synchronize the method; it would be best if it didn't return until the instance has really started.

Reply:
Instance?  This method is static.
I'll change both this and the UPT to have the return inside the synchronized block